dozer-cache/src/cache/lmdb/cache/main_environment/operation_log/mod.rs

```diff
@@ -182,12 +183,12 @@ impl OperationLog {
                     ),
                     Some(metadata) => {
                         let metadata = metadata.borrow();
-                        if metadata.insert_operation_id.is_some() {
-                            // Primary key collision.
-                            return Ok(None);
-                        } else {
+                        if metadata.insert_operation_id.is_none() || upsert_on_collision {
```

Comment:
I think if there's collision and `upsert_on_collision` is `true`, it'll panic on line 221. Maybe add a test case?